docs/src/developing/clients/jsonrpc-api.md

```diff
@@ -122,18 +122,33 @@ Requests can be sent in batches by sending an array of JSON-RPC request objects
 
 ## Configuring State Commitment
 
-Solana nodes choose which bank state to query based on a commitment requirement
-set by the client. Clients may specify either:
-
-- `"max"` - the node will query the most recent block confirmed by supermajority of the cluster as having reached
-maximum lockout.
-- `"root"` - the node will query the most recent block having reached maximum lockout on this node.
+For preflight checks and transaction processing, Solana nodes choose which bank
+state to query based on a commitment requirement set by the client. The
+commitment describes how finalized a block is at some point in time.  When
+querying the ledger state, it's recommended to use lower levels of commitment
+to report progress and higher levels to ensure the state will not be rolled back.
+
+In descending order of commitment (most finalized to least finalized), clients
+may specify:
+
+- `"max"` - the node will query the most recent block confirmed by supermajority
+of the cluster as having reached maximum lockout, meaning the cluster has
+recognized this block as finalized
+- `"root"` - the node will query the most recent block having reached maximum
+lockout on this node, meaning the node has recognized this block as finalized
 - `"singleGossip"` - the node will query the most recent block that has been voted on by supermajority of the cluster.
   - It incorporates votes from gossip and replay.
   - It does not count votes on descendants of a block, only direct votes on that block.
   - This confirmation level also upholds "optimistic confirmation" guarantees in
     release 1.3 and onwards.
-- `"recent"` - the node will query its most recent block.
+- `"recent"` - the node will query its most recent block.  Note that the state at
+this slot is not derived from a finalized block.
```

Comment:
```suggestion
- `"recent"` - the node will query its most recent block.  Note that the block
may not be complete.
```

I think this is what you mean, right?
Finalized means reached max lockout. While it's also true that recent banks have not been finalized, I don't think that's the distinction you were making. Do I have that right?

Reply:
That's much clearer, yes